pallets/liquid-staking/src/lib.rs

```diff
@@ -1732,6 +1765,75 @@ pub mod pallet {
             Ok(())
         }
 
+        #[require_transactional]
+        fn do_fast_match_unstake(unstaker: &T::AccountId) -> DispatchResult {
+            FastUnstakeRequests::<T>::try_mutate_exists(unstaker, |b| -> DispatchResult {
+                if let Some(request_amount) = b.take() {
+                    let available_amount = Self::matching_pool().total_stake_amount.free()?;
+                    let actual_amount = available_amount.min(request_amount);
+
+                    if !actual_amount.is_zero() {
+                        let fee = T::MatchingPoolFastUnstakeFee::get()
+                            .checked_mul_int(actual_amount)
+                            .ok_or(ArithmeticError::Overflow)?;
+                        let transfer_amount = actual_amount
```

Comment:
we have too many amounts util here, I suggest that we limit to:

- amount : unstake_amount * exchangeRate
- matching_fees : fee amount
- receive_amount: `(amount - matching_fees).min(available_amount)`
- matched